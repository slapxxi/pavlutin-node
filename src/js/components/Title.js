import React, { Children } from 'react';
import { withClassName } from './HOC';


function Title({ children, className }) {
  if (Children.count(children) === 1) {
    return <h1 className={className}>{children}</h1>;
  }
  return (
    <hgroup className={className}>
      { Children.map(children, (child, i) => {
        if (i === 0) {
          return React.cloneElement(child, { className: 'title__main' });
        }
        return React.cloneElement(child, { className: 'title__secondary' });
      })}
    </hgroup>
  );
}

export default withClassName('title')(Title);
