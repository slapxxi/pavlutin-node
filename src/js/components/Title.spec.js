import React from 'react';
import render from 'react-test-renderer';
import Title from './Title';

it('renders heading', () => {
  const tree = render.create(<Title>Test</Title>);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders group of headings', () => {
  const tree = render.create(
    <Title>
      <h1>Hello</h1>
      <h2>World</h2>
    </Title>,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('accepts class name', () => {
  const tree = render.create(<Title className="test">Test</Title>);
  expect(tree.toJSON()).toMatchSnapshot();
});
