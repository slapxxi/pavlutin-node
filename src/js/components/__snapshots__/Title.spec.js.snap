// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts class name 1`] = `
<h1
  className="title test"
>
  Test
</h1>
`;

exports[`renders group of headings 1`] = `
<hgroup
  className="title"
>
  <h1
    className="title__main"
  >
    Hello
  </h1>
  <h2
    className="title__secondary"
  >
    World
  </h2>
</hgroup>
`;

exports[`renders heading 1`] = `
<h1
  className="title"
>
  Test
</h1>
`;
